app/src/pages/[resourceType]/[id].tsx

```diff
@@ -28,6 +31,45 @@ export default function ResourceIDPage({ jsonData }: InferGetServerSidePropsType
     /* eslint-enable @typescript-eslint/no-explicit-any */
   }, []);
 
+  const [loadingIconVisible, setLoadingIconVisible] = useState(false);
```

Comment:
I tested this a bit, and we actually want to use the `isFetching` property on `useQuery` rather than `isLoading`. This is because we set enabled to false, so what we actually want is for the loading to occur when the fetching is happening, not when dataRequirements is not populated.

Reply:
That was a great suggestion, thank you! I implemented this and it worked :)